### PR TITLE
Rename md to remark and hast to rehype

### LINKED ISSUE #314: Rename mdPlugins/hastPlugins to remarkPlugins/rehypePlugins
https://github.com/mdx-js/mdx/pull/468

### PR DESCRIPTION
**Pull request by alexandernanberg**
Fixes #314

Account for some changes in MDX@1.0

This is a breaking change so we might want to allow the old way to still work?

### CODE REVIEW

**Comment by johno** [comment]
Yeah, I think we should follow similarly to how we're handling it in MDX core, automatically aliasing the argument and issuing a warning.

**Comment by alexandernanberg** [comment]
It should work now, although I'm not that happy with my solution. Is there any better way to do it? I thought we could reuse the warnings from MDX so we don't have to create our own.

File: packages/gatsby-mdx/utils/gen-mdx.js [hunk 1]
```diff
@@ -97,23 +97,30 @@ module.exports = async function genMDX({
...
-    gatsbyRemarkPlugins: options.gatsbyRemarkPlugins,
-    mdxNode: node,
-    //          files,
-    getNode,
-    getNodes,
-    reporter,
-    cache,
-    pathPrefix
-  });
+  const gatsbyRemarkPluginsAsremarkPlugins = await getSourcePluginsAsRemarkPlugins(
```
Comment (johno) [inline packages/gatsby-mdx/utils/gen-mdx.js]:
```suggestion
  const gatsbyRemarkPluginsAsRemarkPlugins = await getSourcePluginsAsRemarkPlugins(
```

File: packages/gatsby-mdx/utils/gen-mdx.js [hunk 2]
```diff
@@ -97,23 +97,30 @@ module.exports = async function genMDX({
...
+      pathPrefix
+    }
+  );
 
   debug("running mdx");
   let code = await mdx(content, {
     ...options,
-    mdPlugins: options.mdPlugins.concat(gatsbyRemarkPluginsAsMDPlugins)
+    remarkPlugins: options.remarkPlugins.concat(
+      gatsbyRemarkPluginsAsremarkPlugins
```
Comment (johno) [inline packages/gatsby-mdx/utils/gen-mdx.js]:
```suggestion
      gatsbyRemarkPluginsAsRemarkPlugins
```

**Comment by ChristopherBiscardi** [comment]
merged in https://github.com/ChristopherBiscardi/gatsby-mdx/commit/a009d7d401615a79e1e637ed3c0a9a523179c9d3

**Pull request closed without merging**